architecture-model-generator/modules/architecture-model-generator-core/src/main/java/io/ballerina/architecturemodelgenerator/core/generators/entrypoint/nodevisitors/FunctionEntryPointMemberNodeVisitor.java

```diff
@@ -82,9 +83,12 @@ public void visit(VariableDeclarationNode variableDeclarationNode) {
                     boolean isClientClass = referredClassSymbol.qualifiers().stream()
                             .anyMatch(qualifier -> qualifier.equals(Qualifier.CLIENT));
                     if (isClientClass) {
-                        String serviceId = getServiceAnnotation(variableDeclarationNode.annotations(),
-                                filePath.toString()).getId();
-                        Dependency dependency = new Dependency(serviceId,
+                        DisplayAnnotation displayAnnotation =
+                                getServiceAnnotation(variableDeclarationNode.annotations(), filePath.toString());
```

Comment:
shouldn't the display annotation be optional? or do we get null in that case?

Reply:
No, regardless of the values of parameters, we always get `DisplayAnnotation` object as a result here.